packages/client/src/runtime/core/model/applyModel.ts

```diff
@@ -34,29 +34,45 @@ const fluentProps = [
 ] as const
 const aggregateProps = ['aggregate', 'count', 'groupBy'] as const
 
+// used to retrieve the parent client from a model
+const MODEL_PARENT_CLIENT = Symbol.for('prisma.client.model.parent')
+
 /**
  * Dynamically creates a model interface via a proxy.
  * @param client to trigger the request execution
  * @param dmmfModelName the dmmf name of the model
  * @returns
  */
 export function applyModel(client: Client, dmmfModelName: string) {
-  const layers: CompositeProxyLayer[] = [
-    modelActionsLayer(client, dmmfModelName),
-    modelMetaLayer(dmmfModelName),
-    fieldsPropertyLayer(client, dmmfModelName),
+  const modelExtensions = client._extensions.getAllModelExtensions(dmmfModelName) ?? {}
+
+  const getDefaultLayers = (parent = client) => [
+    modelActionsLayer(parent, dmmfModelName),
+    fieldsPropertyLayer(parent, dmmfModelName),
+    addProperty('name', () => dmmfModelName),
+    addProperty(MODEL_PARENT_CLIENT, () => parent),
   ]
 
-  const modelExtensions = client._extensions.getAllModelExtensions(dmmfModelName)
-  if (modelExtensions) {
-    layers.push(addObjectProperties(modelExtensions))
+  const model = createCompositeProxy({}, getDefaultLayers())
+  const modelKeys = Object.keys(model)
+
+  // when a model extension override calls its own name, we refer to its parent
+  // eg. calling findUnique in accelerate's own findUnique calls client.findUnique
```

Comment:
´client` here is always the Client how it was before the Accelerate extension was applied, not the fully unmodified one, correct?

Reply:
It's the client with all extensions minus any override to the builtin methods.